include/eve/module/real/algorithm/function/regular/simd/arm/neon/any.hpp

```diff
@@ -13,31 +13,55 @@
 #include <eve/concept/value.hpp>
 #include <eve/detail/implementation.hpp>
 #include <eve/function/bit_cast.hpp>
+#include <eve/function/convert.hpp>
 
 namespace eve::detail
 {
-  template<real_scalar_value T, typename N>
-  EVE_FORCEINLINE bool any_(EVE_SUPPORTS(neon128_), logical<wide<T,N,arm_64_>> const &v0) noexcept
+  template <typename T, typename N, typename C>
+  EVE_FORCEINLINE bool any_arm_impl(logical<wide<T, N, arm_64_>> v0, C const & cond)
   {
-    if constexpr( N::value == 1 )
-    {
-      return v0.get(0);
-    }
+    using l_t = logical<wide<T, N, arm_64_>>;
+    if constexpr ( C::is_complete && !C::is_inverted ) return false;
     else
     {
+      if constexpr ( !C::is_complete ) v0 = v0 && cond.mask(eve::as_<l_t>{});
       using wide_u32 = wide<std::uint32_t, fixed<2>, arm_64_>;
       auto as_uint32 = eve::bit_cast(v0, eve::as_<eve::logical<wide_u32>>{});
 
-      auto m = as_uint32.bits();
-      m = vorr_u32(m, vrev64_u32(m));
-      return static_cast<bool>(m.get(0));
+      if constexpr( sizeof(T) * N() > 4u ) as_uint32 = vpmax_u32(as_uint32, as_uint32);
+      return vget_lane_u32(as_uint32, 0);
     }
   }
 
+  template <typename T, typename N, typename C>
+  EVE_FORCEINLINE bool any_arm_impl(logical<wide<T, N, arm_128_>> v0, C const & cond)
+  {
+    using l_t = logical<wide<T, N, arm_128_>>;
+
+         if constexpr ( C::is_complete && !C::is_inverted ) return false;
+    else if constexpr ( sizeof( T ) >= 2 )
+    {
+      using half_e_t = make_integer_t<sizeof(T) / 2, unsigned>;
+      auto halved = eve::convert(v0, eve::as_<eve::logical<half_e_t>>{});
```

Comment:
are you sure you want convert here and not bit_cast ?
Not even sure we have non-emulated code for this conversion.

Reply:
I'm sure - I want vector narrow conversions: https://developer.arm.com/documentation/dui0472/m/using-neon-support/neon-intrinsics-for-converting-vectors

But not all of them seem to be in place, hmm:
I see `vmovn_u32` but not `vmovn_u16` for example.